### DISCUSSION TITLE
Working model returns 0 objective value when duplicated

### DISCUSSION THREAD

**Discussion by Kahuna-flipper** [opencobra/cobrapy#1380] [Q&A]
Hello, 

I am working on a reconstruction and trying to diagnose a few things. I have a working model which returns a 'normal' objective value. However, when I try extracting all its reactions and metabolites and create a new duplicate model, it does not return a objective value.

Here is the code I used to duplicate the model : 

```python
from cobra import Model, Reaction, Metabolite

test_model = Model('old_model_testing')

for _,row in old_model_metabolites.iterrows():
    metabolite = Metabolite(row['ID'])
    metabolite.name = row['Name']
    metabolite.formula = row['Formula']
    metabolite.compartment = ''
    test_model.add_metabolites([metabolite])

for _,row in old_model_reactions.iterrows():
    reaction = Reaction(row['ID'])
    test_model.add_reactions([reaction])
    reaction.reaction = row['Reaction']
    reaction.name = row['Name']
    reaction.subsystem = ''
    reaction.lower_bound = row['Lower Bound'] 
    reaction.upper_bound = row['Upper Bound']

test_model.objective = 'NewTestGrowth'
```

I am also happy to share the original json file based on which I made the metabolite (old_model_metabolites) and reactions (old_model_reactions) dataframes.

**Comment by Midnighter** [comment]
Could be many things, but I don't see you adding any exchange reactions, i.e., you're missing a medium that defines uptake rates.

**Reply by Kahuna-flipper** [reply]
But all the exchange reactions are included in the reactions dataframe. The model should then just pick whatever exchanges it deems most optimal for the objective function

**Comment by cdiener** [comment]
Looks like all your added reactions are empty. They have no stoichiometry or metabolites includes so the zero flux makes sense there. Usually you would use `model.copy()` to copy a complete model.

**Reply by Kahuna-flipper** [reply]
Usually I would use .copy(), however this was test to see if I could add reactions from multiple sources to create a new model. I did check all the reactions and they seem to be fine, I even tried using build_from_string, which should identify the stoichiometry but the model returns no optimal value.

**Reply by gregmedlock** [reply]
Could you share your code for this and the dataframes from which you are parsing reactions and metabolites? In the example you provided, no association is being created between the metabolites and the reactions, so in your example, you obtained the expected result (no feasible solution).
